resources/scheduled_task.rb

```diff
@@ -83,6 +83,25 @@
 end
 
 action_class do
+  #
+  # not all frequencies in the windows_task resource support random_delay
+  #
+  # @return [boolean]
+  #
+  def frequency_supports_random_delay?
+    %w(once minute hourly daily weekly monthly).include?(new_resource.frequency)
+  end
+
+  #
+  # not all frequencies in the windows_task resource support frequency_modifier
+  #
+  # @return [boolean]
+  #
+  def frequency_supports_frequency_modifier?
+    # these are the only ones that don't
+    !%w(once on_logon onstart on_idle).include?(new_resource.frequency)
```

Comment:
Just checking, but is `onstart` a typo? Does it need to be `on_start` to match the others?

Reply:
Sadly not a typo: https://github.com/chef/chef/blob/master/lib/chef/resource/windows_task.rb#L72